prometheus/metric.go

```diff
@@ -401,11 +401,19 @@ func le(buckets []stats.Value) string {
 		}
 		b = appendFloat(b, valueOf(v))
 	}
+	return unsafeByteSliceToString(b)
+}
 
-	return *(*string)(unsafe.Pointer(&reflect.StringHeader{
-		Data: uintptr(unsafe.Pointer(&b[0])),
-		Len:  len(b),
-	}))
+// This function converts the byte array to a string without additional
+// memory allocation.
+// Source: https://stackoverflow.com/a/66865482 (license: CC BY-SA 4.0)
```

Comment:
TIL there is a license associated with stack overflow code!  Crazy.
https://stackoverflow.com/help/licensing

Reply:
It caused quite the kerfuffle a few years ago.